data-prep-pipeline/data_prep_pipeline/portal_compounds/transform_portal_compounds.py

```diff
@@ -0,0 +1,100 @@
+import pandas as pd
+from taigapy import create_taiga_client_v3
+
+from config import portal_compounds_taiga_id
+from datarelease_taiga_permanames import prism_oncref_auc_matrix_taiga_permaname
+from utils import update_taiga
+
+repsdrug_matrix_taiga_id = (
+    "public-non-quarterly-processed-files-8e90.64/repsdrug-matrix"
+)
+repsdrug_auc_matrix_taiga_id = (
+    "public-non-quarterly-processed-files-8e90.64/repsdrug-auc-matrix"
+)
+
+
+def filter_sample_ids(
+    sample_ids: str,
+    repsdrug_matrix: pd.DataFrame,
+    repsdrug_auc: pd.DataFrame,
+    oncrefauc_matrix: pd.DataFrame,
+) -> str:
+    """
+    Filters the SampleIDs based on the following rules:
+    - Keep if it starts with 'GDSC' or 'CTRP'.
+    - Keep if it starts with 'BRD:' and is in repsdrug_matrix or repsdrug_auc.
+    - Keep if the stripped version (without 'BRD:') is in oncrefauc_matrix columns.
+    """
+    kept_ids = []
+
+    for sample_id in sample_ids.split(";"):
+        if sample_id.startswith("GDSC") or sample_id.startswith("CTRP"):
+            kept_ids.append(sample_id)
+        elif sample_id.startswith("BRD"):
+            # Check in repsdrug_matrix and repsdrug_auc
+            if sample_id in repsdrug_matrix.index or sample_id in repsdrug_auc.index:
+                kept_ids.append(sample_id)
+            else:
+                # Strip "BRD:" and check in oncrefauc_matrix columns
+                stripped_id = sample_id.replace("BRD:", "")
+                if stripped_id in oncrefauc_matrix.columns:
+                    kept_ids.append(sample_id)
+
+    return ";".join(kept_ids)
+
+
+def filter_portal_compounds(
+    df: pd.DataFrame,
+    repsdrug_matrix: pd.DataFrame,
+    repsdrug_auc: pd.DataFrame,
+    oncrefauc_matrix: pd.DataFrame,
+) -> pd.DataFrame:
+    """
+    Filters the portal_compounds DataFrame by keeping only valid SampleIDs according to the
+    specified rules. Rows with empty SampleIDs are removed.
+    """
+    # Use apply to filter SampleIDs
+    df["SampleIDs"] = df["SampleIDs"].apply(
```

Comment:
A best practice to keep in mind in the future:

The way this function takes in a dataframe and returns a dataframe, means I wouldn't expect this function to _mutate_ df. 

Many times I'll do an initial `df = df.copy()` at the start of a function if I want to mutate things. This will ensure the `df` that was passed in will be unaltered by the end of the function.